tests/virt_autotest/virt_autotest_base.pm

```diff
@@ -152,23 +163,27 @@ sub run_test {
         $timeout = 300;
     }
 
-    my $test_cmd      = $self->get_script_run();
-    my $script_output = $self->execute_script_run($test_cmd, $timeout);
-
-    if ($add_junit_log_flag eq "yes") {
-        $self->add_junit_log($script_output);
+    my $test_cmd = $self->get_script_run();
+    if (check_var('ARCH', 's390x')) {
+       $self->svirt_cmd ("$test_cmd");
+    }
+    else {
+       my $script_output = $self->execute_script_run($test_cmd, $timeout);
+       if ($add_junit_log_flag eq "yes") {
+           $self->add_junit_log($script_output);
+       }
+
+       if ($assert_pattern) {
+           unless ($script_output =~ /$assert_pattern/m) {
+           assert_script_run("grep -E \"$assert_pattern\" $log_dir -r || zcat /tmp/$compressed_log_name.tar.gz | grep -aE \"$assert_pattern\"");
+           }
+        }
     }
 
     if ($upload_virt_log_flag eq "yes") {
         upload_virt_logs($log_dir, $compressed_log_name);
     }
 
-    if ($assert_pattern) {
-        unless ($script_output =~ /$assert_pattern/m) {
-            assert_script_run("grep -E \"$assert_pattern\" $log_dir -r || zcat /tmp/$compressed_log_name.tar.gz | grep -aE \"$assert_pattern\"");
-        }
-    }
```

Comment:
if ($assert_pattern) branch should be executed before if ($upload_virt_log_flag eq "yes"), or there will be missing logs when assert fails.